pkg/oci/skopeo.go

```diff
@@ -1,45 +1,100 @@
 package oci
 
 import (
-	"os"
-	"os/exec"
+    "context"
+    "fmt"
+    "os"
+    "github.com/containers/image/v5/copy"
+    "github.com/containers/image/v5/signature"
+    "github.com/containers/image/v5/transports/alltransports"
 )
 
 // LoadDocker loads the image to the docker daemon
 func LoadDocker(daemon, dir, imageName string) error {
-	cmd := exec.Command("nix", "run", "nixpkgs#skopeo", "--", "copy", "--insecure-policy", "--dest-daemon-host="+daemon, "dir:"+dir, "docker-daemon:"+imageName)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err := cmd.Run()
-	if err != nil {
-		return err
-	}
-
-	return nil
+    ctx := context.Background()
+
+    // Parsed the source and destination image references
+    srcRef, err := alltransports.ParseImageName("dir:" + dir)
+    if err != nil {
+        return fmt.Errorf("parsing source image reference: %w", err)
+    }
+    destRef, err := alltransports.ParseImageName("docker-daemon:" + imageName)
+    if err != nil {
+        return fmt.Errorf("parsing destination image reference: %w", err)
+    }
+
+    // Created a policy context
+    policyContext, err := signature.NewPolicyContext(&signature.Policy{})
+    if err != nil {
+        return fmt.Errorf("creating policy context: %w", err)
+    }
+    defer policyContext.Destroy()
+
+    // Copied the image
+    _, err = copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
+        ReportWriter: os.Stdout,
+    })
+    if err != nil {
+        return fmt.Errorf("copying image: %w", err)
+    }
+
+    return nil
 }
 
-// LoadPodman loads the image to the poadman
+// LoadPodman loads the image to the podman
 func LoadPodman(dir, imageName string) error {
-	cmd := exec.Command("nix", "run", "nixpkgs#skopeo", "--", "copy", "--insecure-policy", "dir:"+dir, "containers-storage:"+imageName)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err := cmd.Run()
-	if err != nil {
-		return err
-	}
-
-	return nil
+    ctx := context.Background()
+
+    srcRef, err := alltransports.ParseImageName("dir:" + dir)
+    if err != nil {
+        return fmt.Errorf("parsing source image reference: %w", err)
+    }
+    destRef, err := alltransports.ParseImageName("containers-storage:" + imageName)
+    if err != nil {
+        return fmt.Errorf("parsing destination image reference: %w", err)
+    }
+
+    policyContext, err := signature.NewPolicyContext(&signature.Policy{})
+    if err != nil {
+        return fmt.Errorf("creating policy context: %w", err)
+    }
+    defer policyContext.Destroy()
+
+    _, err = copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
+        ReportWriter: os.Stdout,
```

Comment:
instead of streaming to `os.Stdout` , can we accept `io.Writer` as an arg to these functions and use that here?
`cmd/` can then use `os.Stdout`.

Reply:
I'll do that !